### PR TITLE
fix: `database_url` is optional in env

### LINKED ISSUE #18: fix: `database_url` is optional in env
Make `database_url` a required env var

### PR DESCRIPTION
**Pull request by thaodt**
# Description

make `database_url` is required in env.

Resolves #18 

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/env.rs [hunk 1]
```diff
@@ -36,6 +36,10 @@ impl Config {
         tracing::Level::from_str(self.log_level.as_str()).expect("Invalid log level")
     }
 
+    pub fn database_url(&self) -> &str {
```
Comment (HarryET) [inline src/env.rs]: Not needed.

File: src/main.rs [hunk 2]
```diff
@@ -39,14 +39,9 @@ async fn main() -> error::Result<()> {
...
 
-    let database_url = config
-        .database_url
-        .as_deref()
-        .expect("database url is missing");
-
     let store = PgPoolOptions::new()
         .max_connections(5)
-        .connect(database_url)
+        .connect(config.database_url())
```
Comment (HarryET) [inline src/main.rs]: Access config instead

**Review by HarryET — CHANGES_REQUESTED** [review]

File: src/env.rs (hunk 1 above)
Comment (thaodt) [reply src/env.rs]: removed.

File: src/main.rs (hunk 2 above)
Comment (thaodt) [reply src/main.rs]: > Access config instead

updated.

File: src/env.rs [hunk 3]
```diff
@@ -64,6 +65,10 @@ fn default_log_level() -> String {
     "WARN".to_string()
 }
 
+fn default_database_url() -> String {
```
Comment (HarryET) [inline src/env.rs]: We don't need a default, if its not there it can panic on startup

Comment (HarryET) [inline src/env.rs:68]: We don't need a default, if its not there it can panic on startup

**Review by HarryET — CHANGES_REQUESTED** [review]

File: src/env.rs (hunk 3 above)
Comment (thaodt) [reply src/env.rs]: @HarryET hmm this change served for `tests`, have you rerun the tests on my PR? Do you want me provide a dummy string instead? If i don't set it, tests will fail.

**Comment by thaodt** [comment]
hi @HarryET , already ping you in conversation above but maybe you didn't notice. Therefore, will add a comment here for details.
This latest commit required changes in `tests` to make it passed. Please double-check.

**Comment by HarryET** [comment]
> hi @HarryET , already ping you in conversation above but maybe you didn't notice. Therefore, will add a comment here for details. This latest commit required changes in `tests` to make it passed. Please double-check.

Hi @thaodt, sorry for the delay been busy. Looks good will have a proper look soon and try merge it!

File: src/env.rs (hunk 3 above)
Comment (HarryET) [reply src/env.rs]: Yeah I know tests will fail, just update the tests

**Review by HarryET — APPROVED** [review]

**Pull request merged**